homeassistant/components/anthropic/conversation.py

```diff
@@ -266,34 +366,58 @@ async def _async_handle_message(
         system = chat_log.content[0]
         if not isinstance(system, conversation.SystemContent):
             raise TypeError("First message must be a system message")
-        messages = [_convert_content(content) for content in chat_log.content[1:]]
+        messages = _convert_content(chat_log.content[1:])
 
         client = self.entry.runtime_data
 
+        thinking_budget = options.get(CONF_THINKING_BUDGET, RECOMMENDED_THINKING_BUDGET)
+        model = options.get(CONF_CHAT_MODEL, RECOMMENDED_CHAT_MODEL)
+
         # To prevent infinite loops, we limit the number of iterations
         for _iteration in range(MAX_TOOL_ITERATIONS):
+            if model in THINKING_MODELS and thinking_budget >= MIN_THINKING_BUDGET:
+                model_args = {
+                    "model": model,
+                    "messages": messages,
+                    "tools": tools or NOT_GIVEN,
+                    "max_tokens": options.get(CONF_MAX_TOKENS, RECOMMENDED_MAX_TOKENS),
+                    "system": system.content,
+                    "stream": True,
+                    "thinking": ThinkingConfigEnabledParam(
+                        type="enabled", budget_tokens=thinking_budget
+                    ),
+                }
+            else:
+                model_args = {
+                    "model": model,
+                    "messages": messages,
+                    "tools": tools or NOT_GIVEN,
```

Comment:
Define a dict above the if statement with all the common keys.

Reply:
Done